executor/executor_darwin.h

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+#include <math.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+
+// FIXME(HerrSpace): As executor is written in C++, we need to make this patch:
+// -struct ksancov_trace *trace = (void *)mc.ptr; into
+// +struct ksancov_trace *trace = (ksancov_trace *)mc.ptr;
+// twice to make this header compile. This used to be C++ friendly in Catalina,
+// but was broken in xnu source drop 7195.50.7.100.1.
+#include <ksancov.h>
+
+static void os_init(int argc, char** argv, void* data, size_t data_size)
+{
```

Comment:
I think we also need to set is_kernel_64_bit=false somewhere here so that write_coverage_signal will treat coverage as an array of uint32's.

Reply:
Uh. I thought my smarty-pants comment in executor.cc would remove the necessity for this, but after looking at it again I agree, we do need this after all. Added :)